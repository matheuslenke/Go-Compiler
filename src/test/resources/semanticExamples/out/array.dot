digraph {
graph [ordering="out"];
node0[label="source_file"];
node1[label="(func) testeArrays@0"];
node2[label="parameters"];
node1 -> node2;
node3[label="block"];
node4[label="var_list"];
node5[label="(array) a@1"];
node6[label="array_length"];
node7[label="(int) int@2"];
node6 -> node7;
node5 -> node6;
node8[label="(string) element_type"];
node5 -> node8;
node4 -> node5;
node3 -> node4;
node9[label="var_list"];
node10[label="(array) b@2"];
node11[label="array_length"];
node12[label="(int) int@2"];
node11 -> node12;
node10 -> node11;
node13[label="(int) element_type"];
node10 -> node13;
node9 -> node10;
node3 -> node9;
node14[label="var_list"];
node15[label="(array) c@3"];
node16[label="array_length"];
node17[label="(int) int@2"];
node16 -> node17;
node15 -> node16;
node18[label="(float) element_type"];
node15 -> node18;
node14 -> node15;
node3 -> node14;
node19[label="var_list"];
node20[label="(array) d@4"];
node21[label="array_length"];
node22[label="(int) int@2"];
node21 -> node22;
node20 -> node21;
node23[label="(float) element_type"];
node20 -> node23;
node19 -> node20;
node3 -> node19;
node24[label="var_list"];
node25[label="(array) e@5"];
node26[label="array_length"];
node27[label="(int) int@2"];
node26 -> node27;
node25 -> node26;
node28[label="(bool) element_type"];
node25 -> node28;
node24 -> node25;
node3 -> node24;
node29[label="var_list"];
node30[label="="];
node31[label="(int) IntVar@6"];
node30 -> node31;
node32[label="(int) int@2"];
node30 -> node32;
node29 -> node30;
node3 -> node29;
node33[label="assign_list_node"];
node34[label="="];
node35[label="(array) b@2"];
node36[label="(int) index"];
node37[label="(int) int@0"];
node36 -> node37;
node35 -> node36;
node34 -> node35;
node38[label="(int) IntVar@6"];
node34 -> node38;
node33 -> node34;
node3 -> node33;
node39[label="assign_list_node"];
node40[label="="];
node41[label="(int) IntVar@6"];
node40 -> node41;
node42[label="(array) b@2"];
node43[label="(int) index"];
node44[label="(int) int@0"];
node43 -> node44;
node42 -> node43;
node40 -> node42;
node39 -> node40;
node3 -> node39;
node45[label="var_list"];
node46[label="="];
node47[label="(float) float32Var@7"];
node46 -> node47;
node48[label="(float) real@5.0"];
node46 -> node48;
node45 -> node46;
node3 -> node45;
node49[label="assign_list_node"];
node50[label="="];
node51[label="(array) c@3"];
node52[label="(int) index"];
node53[label="(int) int@0"];
node52 -> node53;
node51 -> node52;
node50 -> node51;
node54[label="(float) float32Var@7"];
node50 -> node54;
node49 -> node50;
node3 -> node49;
node55[label="assign_list_node"];
node56[label="="];
node57[label="(float) float32Var@7"];
node56 -> node57;
node58[label="(array) c@3"];
node59[label="(int) index"];
node60[label="(int) int@0"];
node59 -> node60;
node58 -> node59;
node56 -> node58;
node55 -> node56;
node3 -> node55;
node61[label="var_list"];
node62[label="="];
node63[label="(float) float64Var@8"];
node62 -> node63;
node64[label="(float) real@10.0"];
node62 -> node64;
node61 -> node62;
node3 -> node61;
node65[label="assign_list_node"];
node66[label="="];
node67[label="(array) d@4"];
node68[label="(int) index"];
node69[label="(int) int@0"];
node68 -> node69;
node67 -> node68;
node66 -> node67;
node70[label="(float) float64Var@8"];
node66 -> node70;
node65 -> node66;
node3 -> node65;
node71[label="assign_list_node"];
node72[label="="];
node73[label="(float) float64Var@8"];
node72 -> node73;
node74[label="(array) d@4"];
node75[label="(int) index"];
node76[label="(int) int@0"];
node75 -> node76;
node74 -> node75;
node72 -> node74;
node71 -> node72;
node3 -> node71;
node77[label="var_list"];
node78[label="="];
node79[label="(bool) trueVar@9"];
node78 -> node79;
node80[label="(bool) bool@true"];
node78 -> node80;
node77 -> node78;
node3 -> node77;
node81[label="var_list"];
node82[label="="];
node83[label="(bool) falseVar@10"];
node82 -> node83;
node84[label="(bool) bool@false"];
node82 -> node84;
node81 -> node82;
node3 -> node81;
node85[label="assign_list_node"];
node86[label="="];
node87[label="(array) e@5"];
node88[label="(int) index"];
node89[label="(int) int@0"];
node88 -> node89;
node87 -> node88;
node86 -> node87;
node90[label="(bool) trueVar@9"];
node86 -> node90;
node85 -> node86;
node3 -> node85;
node91[label="assign_list_node"];
node92[label="="];
node93[label="(array) e@5"];
node94[label="(int) index"];
node95[label="(int) int@1"];
node94 -> node95;
node93 -> node94;
node92 -> node93;
node96[label="(bool) bool@true"];
node92 -> node96;
node91 -> node92;
node3 -> node91;
node97[label="assign_list_node"];
node98[label="="];
node99[label="(bool) falseVar@10"];
node98 -> node99;
node100[label="(array) e@5"];
node101[label="(int) index"];
node102[label="(int) int@1"];
node101 -> node102;
node100 -> node101;
node98 -> node100;
node97 -> node98;
node3 -> node97;
node103[label="var_list"];
node104[label="="];
node105[label="(float) addFloat@11"];
node104 -> node105;
node106[label="(float) +"];
node107[label="(array) c@3"];
node108[label="(int) index"];
node109[label="(int) int@0"];
node108 -> node109;
node107 -> node108;
node106 -> node107;
node110[label="(array) c@3"];
node111[label="(int) index"];
node112[label="(int) int@1"];
node111 -> node112;
node110 -> node111;
node106 -> node110;
node104 -> node106;
node103 -> node104;
node3 -> node103;
node113[label="var_list"];
node114[label="="];
node115[label="(float) mulFloat@12"];
node114 -> node115;
node116[label="(float) *"];
node117[label="(array) c@3"];
node118[label="(int) index"];
node119[label="(int) int@0"];
node118 -> node119;
node117 -> node118;
node116 -> node117;
node120[label="(array) c@3"];
node121[label="(int) index"];
node122[label="(int) int@1"];
node121 -> node122;
node120 -> node121;
node116 -> node120;
node114 -> node116;
node113 -> node114;
node3 -> node113;
node123[label="var_list"];
node124[label="="];
node125[label="(bool) relFloat@13"];
node124 -> node125;
node126[label="(bool) <"];
node127[label="(array) c@3"];
node128[label="(int) index"];
node129[label="(int) int@0"];
node128 -> node129;
node127 -> node128;
node126 -> node127;
node130[label="(array) c@3"];
node131[label="(int) index"];
node132[label="(int) int@1"];
node131 -> node132;
node130 -> node131;
node126 -> node130;
node124 -> node126;
node123 -> node124;
node3 -> node123;
node1 -> node3;
node0 -> node1;
}
