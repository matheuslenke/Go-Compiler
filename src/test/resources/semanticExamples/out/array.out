[INFO] Scanning for projects...
[INFO] 
[INFO] ----------------< br.ufes.edu.compiladores:go-compiler >----------------
[INFO] Building Compilador de GO 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- exec-maven-plugin:3.0.0:java (default-cli) @ go-compiler ---
PARSE SUCCESSFUL



Strings table:




Variables table:
Entry 0 -- name: testeArrays, line: 3, type: func, scopeName: global
Entry 1 -- name: a, line: 5, type: array, subtype: string, scopeName: testeArrays
Entry 2 -- name: b, line: 6, type: array, subtype: int, scopeName: testeArrays
Entry 3 -- name: c, line: 7, type: array, subtype: float, scopeName: testeArrays
Entry 4 -- name: d, line: 8, type: array, subtype: float, scopeName: testeArrays
Entry 5 -- name: e, line: 9, type: array, subtype: bool, scopeName: testeArrays
Entry 6 -- name: IntVar, line: 11, type: int, scopeName: testeArrays
Entry 7 -- name: float32Var, line: 15, type: float, scopeName: testeArrays
Entry 8 -- name: float64Var, line: 19, type: float, scopeName: testeArrays
Entry 9 -- name: trueVar, line: 23, type: bool, scopeName: testeArrays
Entry 10 -- name: falseVar, line: 24, type: bool, scopeName: testeArrays
Entry 11 -- name: addFloat, line: 29, type: float, scopeName: testeArrays
Entry 12 -- name: mulFloat, line: 30, type: float, scopeName: testeArrays
Entry 13 -- name: relFloat, line: 31, type: bool, scopeName: testeArrays




[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.849 s
[INFO] Finished at: 2022-02-24T20:53:38-03:00
[INFO] ------------------------------------------------------------------------
